demos/python_demos/multi_camera_multi_person_tracking/utils/network_wrappers.py

```diff
@@ -81,3 +105,215 @@ def forward(self, batch):
+
+            self.height = None
+            self.width = None
+            if size is not None:
+                self.height, self.width = size
+
+        def get_scale(self, image_size):
+            if self.height is not None:
+                scale_x, scale_y = self.width / image_size[1], self.height / image_size[0]
+            elif self.height_max is not None:
+                im_scale = min(self.height_max / image_size[0], self.width_max / image_size[1])
+                scale_x, scale_y = im_scale, im_scale
+            else:
+                im_scale = min(self.short_side_max / min(image_size), self.long_side_max / max(image_size))
+                scale_x, scale_y = im_scale, im_scale
+            return scale_x, scale_y
+
+        def __call__(self, sample):
+            image_size = sample['image'].shape[:2]
+            scale_x, scale_y = self.get_scale(image_size)
+
+            # Resize image.
+            sample['image'] = cv2.resize(sample['image'], None, fx=scale_x, fy=scale_y)
+            h, w = sample['image'].shape[:2]
+
+            # Resize boxes.
+            if 'gt_boxes' in sample:
+                sample['gt_boxes'] *= [scale_x, scale_y, scale_x, scale_y]
+                sample['gt_boxes'] = np.clip(sample['gt_boxes'], 0, [w - 1, h - 1, w - 1, h - 1])
+
+            # Resize masks.
+            if 'gt_masks' in sample:
+                sample['gt_masks'] = [[np.clip(part * [scale_x, scale_y], 0, [w - 1, h - 1]) for part in obj]
+                                      for obj in sample['gt_masks']]
+
+            return sample
+
+    class Normalize(object):
+        def __init__(self, mean, std):
+            self.mean = mean
+            self.std = std
+
+        def __call__(self, sample):
+            sample['image'] = sample['image'].astype('float32').transpose(2, 0, 1)
+            sample['image'][:, :, 0] = (sample['image'][:, :, 0] - self.mean[0]) / self.std[0]
+            sample['image'][:, :, 1] = (sample['image'][:, :, 1] - self.mean[1]) / self.std[1]
+            sample['image'][:, :, 2] = (sample['image'][:, :, 2] - self.mean[2]) / self.std[2]
+            return sample
+
+
+class DetectionsFromFileReader(DetectorInterface):
+    """Read detection from *.json file.
+    Format of the file should be:
+    [
+        {'frame_id': N,
+         'scores': [score0, score1, ...],
+         'boxes': [[x0, y0, x1, y1], [x0, y0, x1, y1], ...]},
+        ...
+    ]
+    """
+
+    def __init__(self, input_files, score_thresh):
+        self.input_files = input_files
+        self.score_thresh = score_thresh
+        self.detections = []
+        for input_file in input_files:
+            log.info('Loading {}'.format(input_file))
+            with open(input_file) as f:
+                detections = json.load(f)
+                detections_dict = {}
+                for det in detections:
+                    detections_dict[det['frame_id']] = {'boxes': det['boxes'], 'scores': det['scores']}
```

Comment:
I think the format of files for `--history_file` and `--detections` should be aligned. It will be possible to generate files by `--history_file` and provide the same files with `--detections` later.

Reply:
Actually purpose of these two files is different. `--detections` has a simple format to make it possible create file with detected bounding boxes in easy way and use it instead some detector that helps execute demo faster or, for example, check quality of your person ReID-model. I suppose that this file must created by everyone in the simplest way. `--history_file` firstly consists of data obtained from all videos which were used in the demo, then the data is grouped by `IDs` of persons. We can't transfer `--history_file` to `--detections` format, `--detections` to `--history_file` is possible but I think for user it is not useful.

Reply:
> this file must created by everyone in the simplest way

This doesn't seem to be simple. Where does the file for `--detections` come from, maybe there a script to build the file or it can be found in some dataset. Is it possible to clarify/simplify this? There is a note in Readme, but I think it is not very useful because it still requires saving detections somehow.

Reply:
OK, you won. I have added option to save detections.

Reply:
This fails for me at `json.dump(data, outfile)`.

Reply:
I suggest replacing `entry['scores'].append(det[1])` with `entry['scores'].append(det[1].item())`.

Reply:
> This fails for me at `json.dump(data, outfile)`.

Could you please share a full command that leads to the error?

Reply:
> I suggest replacing `entry['scores'].append(det[1])` with `entry['scores'].append(det[1].item())`.

Please clarify why. `det` is always `list` or `tuple` and `det[1]` is always `float32`, method `item()` failed for `tuple`.

Reply:
There is the command:
`python3 open_model_zoo/demos/python_demos/multi_camera_multi_person_tracking/multi_camera_multi_person_tracking.py --m_reid /mnt/h/d/downloader/intel/person-reidentification-retail-0200/FP32/person-reidentification-retail-0200.xml -d CPU -i /home/vzlobin/Videos/cr/test_video_6.avi /home/vzlobin/Videos/cr/test_video_6.avi --config open_model_zoo/demos/python_demos/multi_camera_multi_person_tracking/config.py --m_segmentation /mnt/h/d/downloader/intel/instance-segmentation-security-0050/FP32/instance-segmentation-security-0050.xml --save_detections detections.txt`
It fails after pressing Esc.

Numpy's array cant be saved to json.

Reply:
Fixed.